contracts/hydro/src/contract.rs

```diff
@@ -1700,8 +1708,60 @@ pub fn query_all_user_lockups(
         })
         .collect();
 
+    // enrich lockups with some info per tranche
+    let lockups_with_per_tranche_info: Vec<LockupWithPerTrancheInfo> = locks_with_power
+        .iter()
+        .map(|lock| {
+            // iterate all tranches
+            let tranche_infos = tranche_ids
+                .iter()
+                .map(|tranche_id| {
+                    // add which proposal the lock voted for
+                    let voted_for_proposal: Option<u64> = VOTE_MAP
+                        .may_load(
+                            deps.storage,
+                            (
+                                (current_round_id, *tranche_id),
+                                converted_addr.clone(),
+                                lock.lock_entry.lock_id,
+                            ),
+                        )
+                        .unwrap_or(None)
+                        .map(|vote| vote.prop_id);
+
+                    let next_round_voting_allowed: u64 = if voted_for_proposal.is_some() {
+                        // if the proposal voted in this round, we ignore the VOTING_ALLOWED_ROUND map,
+                        // since it just contains the future information on when the lockup will be able to vote again
+                        // if it doesn't change the vote, but it can vote in the current round either way
+                        current_round_id
+                    } else {
+                        // if the lockup has not voted in this round, VOTING_ALLOWED_ROUND does contain
+                        // current information on whether the votup can vote right now or not
+                        VOTING_ALLOWED_ROUND
+                            .may_load(deps.storage, (*tranche_id, lock.lock_entry.lock_id))
+                            .unwrap_or_default()
```

Comment:
I would prefer skipping entry if `may_load()` returned error (couldn't parse the value). This can be done by using `filter_map()` instead of `map()`.

Reply:
is this comment in the wrong line? I already am doing unwrap_or_default here, same as above

Reply:
No, the line is correct. What I am saying is that `unwrap_or_default()` will return Some(0) if `may_load()` returned Err. I think we should skip that tranche instead.

Reply:
Ok, I see, filter_map in the loop thats enclosing this. that makes sense to me, I adjusted it. please check it out :)